### PR TITLE
Discord mentors permission

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add mentor? method to User
- Set mentor role in Discord
- Add private
- Refactor setting discord roles
- Add test to verify that discord roles are updated when updating mentored tracks

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -344,6 +344,12 @@ def captcha_required? = !github_auth? && Time.current - created_at < 2.days
 
   def flair = super&.to_sym
 
+  def mentor?
+    return true if staff?
+
+    track_mentorships.exists?
```
Comment (iHiD) [inline app/models/user.rb]: I think this should actually be `user.became_mentor_at`

Comment (iHiD) [reply app/models/user.rb]: (There's probably a helper method on `User` already)

Comment (ErikSchierboom) [reply app/models/user.rb]: > I think this should actually be user.became_mentor_at

That does imply that you don't have to be an active mentor, is that correct?

> (There's probably a helper method on User already)

Nope

Comment (iHiD) [reply app/models/user.rb]: Yeah. It might be for people who want advice before starting.

There's a `user.mentor?` you can use (it's defined via roles)

**Comment by iHiD** [comment]
This is red btw.

**Comment by ErikSchierboom** [comment]
Green now!

**Pull request merged**